### DISCUSSION TITLE
How to create a joined table with two shared expressions

### DISCUSSION THREAD

**Discussion by Stewwe1** [TabularEditor/TabularEditor3#389] [🙋♀️TE3 Q&A]
Hello TE community

One quick question:

Maybe I'm just too stupid right now, but how do I create a new table that is to be joined on the basis of two shared expressions?

Creating the shared expressions are no problem, but when copying the M code for the new table into the partition and clicking on full refresh, I would have hoped that it would recreate the columns.

Previously Visual Studio created all the tables for me, but we are just looking to migrate our SSAS model to a PowerBI Data Set and so I am just trying to do everything with the Tabular Editor. Ok, almost everything. I create the power query code with PowerBI.

I am looking forward to your help! :-)

Thanks stewwe

**Comment by otykier** [comment]
Hi @Stewwe1 
The **Full Refresh** option is used when you want to instruct Analysis Services/Power BI to connect to the data source and load data into the table. In other words, it assumes that the table was already set up with the correct columns/data types.

Use the **Update Table Schema...** option when you want to populate the columns on the table based on the partition queries.

**Note:** Depending on the complexity of the queries, this does not always work natively in Tabular Editor, in which case you might need to use [this workaround](https://docs.tabulareditor.com/te3/tutorials/importing-tables.html#updating-table-schema-through-analysis-services).
